R/create_target_file.R

```diff
@@ -19,18 +19,29 @@
 #'                                           dir = 'targets',
 #'                                           mask = fire_box$maskLayer)
 
-create_target_file <- function(cuberast, date, dir, mask = NULL){
-  ## write geotif file
-  dir.create(dir, FALSE)
-  #if statement to mask data in burn area
-  if(!is.null(mask)){
-    #Data masked
-    cuberast %>% gdalcubes::select_time(date) %>% gdalcubes::filter_geom(geom = mask, srs = "EPSG:4326") %>% write_tif(dir = dir)
+create_target_file <- function(cuberast, date, dir=NULL, mask = NULL){
+  
+  target <- cuberast %>%
+    gdalcubes::slice_time(date) 
+  
+  if (!is.null(mask)) {
+    target <- target %>%
+      gdalcubes::filter_geom(geom = mask,
+                             srs = "EPSG:4326")
+  }
+
+  if (!is.null(dir)) {
+    
+    # needed to write geotif to VSI
+    Sys.setenv("CPL_VSIL_USE_TEMP_FILE_FOR_RANDOM_WRITE"="YES") 
+    
+    # doesn't take VSI yet; so convert to stars first instead:
+    # write_tif(target, dir, "lai_recovery_target_") 
     
+    target %>% 
+      stars::st_as_stars() %>%
```

Comment:
This is erroring for me. I assumed related to `CPL_VSIL_USE_TEMP_FILE_FOR_RANDOM_WRITE` but occurs when set to "YES" or "NO"

```
>  target %>% 
+       stars::st_as_stars()
[>                                                 ] 0 %
 *** caught segfault ***
address 0x118, cause 'invalid permissions'

Traceback:
 1: gdalcubes:::gc_exec_worker(j$cube, j$worker_id, j$worker_count,     j$workdir, gdalcubes_options()$ncdf_compression_level)
An irrecoverable exception occurred. R is aborting now ...
[WORKER #0] ## Temporal extent of the cube does not align with dt and has been extended to 2023-05-11/2023-06-09
Error: one or more worker processes failed to compute data cube chunks
```

Reply:
@emmamendelsohn thanks for testing and my mistake!  

I moved the bucket access over to the EFI object store, (see https://github.com/cboettig/modis-lai-forecast/blob/main/R/spat4cast-envars.R ) but forgot to update EFI's object store to recognize the spatial user credentials.  I can add that back now.  Until then I think all `write` or `cp` ops to s3 bucket will fail.

Reply:
@emmamendelsohn ok can you test again? I think it should be fixed to recognize spatial credentials on any `spat4cast-*` buckets on the efi server (web interface at https://minio.ecoforecast.org, S3 API interface at https://data.ecoforecast.org)

Reply:
Still getting the same error. I realized I don't have`EFI_KEY` or `EFI_SECRET` tokens -- how do I get these?

However, I get the error specifically at `stars::st_as_stars()`, so I assume it's something internal failing? I have `stars` 0.6.2 (latest from GitHub) and `GDAL` 3.6.4.

Reply:
Gotcha. Still get the error with with the credentials set, but I must have something set up incorrectly on my end. I'm away for the next two weeks. Perhaps @ddurden or @johnwilliamsmithjr can test the workflow on their ends. Otherwise feel free to merge in. The rendered qmd looks good!

Reply:
ok thanks for the heads up.  My guess is a collision with some credentials you may have set in `~/.aws` (I don't think env vars take precedence over those) but could be something else.  I'll merge now mostly so I can point people to the eco4cast repo copy, but we can also continue to iterate.  

A nice next step I think would have the components we demo here (target generation, forecast generation, scoring) each run as independent GitHub actions.